src/Microsoft.AspNet.Authentication.Google/GoogleAuthenticationHandler.cs

```diff
@@ -30,45 +30,48 @@ public GoogleAuthenticationHandler(HttpClient httpClient)
             graphResponse.EnsureSuccessStatusCode();
             var text = await graphResponse.Content.ReadAsStringAsync();
             var user = JObject.Parse(text);
-
-            var context = new GoogleAuthenticatedContext(Context, Options, user, tokens);
-            var identity = new ClaimsIdentity(
-                Options.ClaimsIssuer,
-                ClaimsIdentity.DefaultNameClaimType,
-                ClaimsIdentity.DefaultRoleClaimType);
-
+            
+            var context = new GoogleAuthenticatedContext(Context, Options, Backchannel, user, tokens)
+            {
+                Properties = properties,
+                Principal = new ClaimsPrincipal(identity)
+            };
+            
             if (!string.IsNullOrEmpty(context.Id))
             {
                 identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, context.Id,
                     ClaimValueTypes.String, Options.ClaimsIssuer));
             }
```

Comment:
A slight tweak here could improve performance.
ClaimIdentity.AddClaim(...) performs an object.ReferenceEquals(claim.Subject, this). If false, then claim.Clone is called.  
If you did identity.AddClaim(..., identity), then claim.Clone would not be called.